src/components/common/TextFieldWithCounter.tsx

```diff
@@ -27,25 +32,33 @@ const TextFieldWithCounter = (
 
   const { maxLen, isCounterHidden, ...other } = props;
 
-  const getCounterClassNames = (): string => {
-    const classNames = [classes.counter];
-    if (maxLen && textLen > maxLen) {
-      classNames.push(classes.error);
-    }
-
-    return classNames.join(' ');
-  };
-
-  const counter = isCounterHidden ? null : (
-    <span className={getCounterClassNames()}>
-      {textLen ? (maxLen ? `${textLen}/${maxLen}` : textLen) : ''}
-    </span>
-  );
+  const counter = isCounterHidden
```

Comment:
This is still a bit unreadable when it is hidden with lot of nested conditional operations but I guess it is a bit better than previously

Reply:
I'll add comments in a different PR or see if I can improve the readability a bit.